src/Elmish/React/Ref.purs

```diff
@@ -0,0 +1,36 @@
+module Elmish.React.Ref
+  ( Ref
+  , callbackRef
+  )
+  where
+
+import Prelude
+
+import Data.Maybe (Maybe(..))
+import Data.Nullable (Nullable)
+import Data.Nullable as Nullable
+import Effect (Effect)
+import Effect.Uncurried (EffectFn1)
+import Elmish.Dispatch ((<?|))
+import Elmish.Foreign (class CanPassToJavaScript)
+import Unsafe.Coerce (unsafeCoerce)
+
+-- | An opaque type representing the type for React `ref` props
+data Ref (el :: Type)
+
+instance CanPassToJavaScript (Ref a)
+
+-- | Turns a callback function (`el -> Effect Unit`) into a `Ref`. The callback
+-- | function should add the `el` parameter to some state.
```

Comment:
Maybe an explanation of parameters and an example would be useful here. The parameters do seem non-trivial.